lisa/combinators/git_bisect_combinator.py

```diff
@@ -0,0 +1,202 @@
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return GitBisectCombinatorSchema
+
+    def _next(self) -> Optional[Dict[str, Any]]:
+        next: Optional[Dict[str, Any]] = None
+        self._process_result()
+        if not self._check_bisect_complete():
+            next = {}
+            next["ref"] = self._get_current_commit_hash()
+        self._result_notifier.increment_iteration()
+        return next
+
+    def _process_result(self) -> None:
+        iteration = self._result_notifier.get_iteration_count()
+        if iteration > 0:
+            results = self._result_notifier.results.get(iteration)
+            if results is not None:
+                if results:
+                    self._bisect_good()
+                else:
+                    self._bisect_bad()
+            else:
+                raise LisaException(
+                    f"Result missing for interation {iteration}, {results}"
+                )
+
+    @with_remote_node
+    def _clone_source(self, node: Node) -> None:
+        # node = self._get_remote_node()
+        node.execute(
+            cmd=f"mkdir -p {SOURCE_PATH}", shell=True, sudo=True, expected_exit_code=0
+        )
+        node.execute(cmd=f"chmod 777 {SOURCE_PATH}", sudo=True, expected_exit_code=0)
+
+        git = node.tools[Git]
+        git.clone(url=self.runbook.repo, cwd=SOURCE_PATH, dir_name=".", timeout=1200)
+
+    def _get_remote_node(self) -> Node:
+        node = quick_connect(self.runbook.connection, "source_node")
+        return node
+
+    @with_remote_node
+    def _start_bisect(self, node: Node) -> None:
+        git = node.tools[Git]
+        git.bisect(cwd=SOURCE_PATH, cmd="start")
+        git.bisect(cwd=SOURCE_PATH, cmd=f"good {self.runbook.good_commit}")
+        git.bisect(cwd=SOURCE_PATH, cmd=f"bad {self.runbook.bad_commit}")
+
+    @with_remote_node
+    def _bisect_bad(self, node: Node) -> None:
+        git = node.tools[Git]
+        git.bisect(cwd=SOURCE_PATH, cmd="bad")
+
+    @with_remote_node
+    def _bisect_good(self, node: Node) -> None:
+        git = node.tools[Git]
+        git.bisect(cwd=SOURCE_PATH, cmd="good")
+
+    @with_remote_node
+    def _check_bisect_complete(self, node: Node) -> bool:
+        git = node.tools[Git]
+        result = git.bisect(cwd=SOURCE_PATH, cmd="log")
+        if any(pattern in result.stdout for pattern in STOP_PATTERNS):
+            return True
+        return False
+
+    @with_remote_node
+    def _get_current_commit_hash(self, node: Node) -> str:
+        git = node.tools[Git]
+        result = git.run("rev-parse HEAD", cwd=SOURCE_PATH, force_run=True, shell=True)
+        return result.stdout
+
+
+class GitBisectResult(notifier.Notifier):
+    @classmethod
+    def type_name(cls) -> str:
+        return ""
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return schema.Notifier
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self._iteration = 0
```

Comment:
The iteration looks not useful much. You can use a result field with triple status like `result: Optional[bool]`. If it's None, it means there is no result receive. True/False present the test result.